synapse_invite_checker/invite_checker.py

```diff
@@ -663,6 +684,21 @@ async def on_create_room(
                     errors.Codes.FORBIDDEN,
                 )
 
+    async def check_login_for_spam(
+        self,
+        user_id: str,
+        device_id: str | None,
+        initial_display_name: str | None,
+        request_info: Collection[tuple[str | None, str]],
+        auth_provider_id: str | None = None,
+    ) -> Literal["NOT_SPAM"] | errors.Codes:
+        # Setting initial permissions is as simple as calling the get_permissions
+        # function. It will be populated based on the related configuration settings.
+        current_perms = await self.permissions_handler.get_permissions(user_id)
+        # Then just save them back. Done deal.
+        await self.permissions_handler.update_permissions(user_id, current_perms)
```

Comment:
I don't think we should write to the database, if the permissions are already set.